cli/src/main.rs

```diff
@@ -1,31 +1,84 @@
 #![allow(dead_code)]
-mod cli;
-mod repl;
-mod report;
 
-use crate::cli::{handle_source, Cli};
-use crate::repl::prompt;
-use clap::Parser;
-use context::source::Source;
-use miette::MietteHandlerOpts;
-use std::io;
+use ariadne::{Color, StdoutFmt};
 use std::ops::Deref;
+use std::path::PathBuf;
 use std::process::exit;
 
-fn main() -> io::Result<()> {
+use clap::Parser;
+
+use crate::cli::{Cli, Configuration};
+use crate::repl::repl;
+use crate::runner::run;
+
+mod cli;
+mod display_bytecode;
+mod formatted_diagnostic;
+mod formatted_parse_error;
+mod repl;
+mod runner;
+mod source_importer;
+
+fn main() {
     let cli = Cli::parse();
 
-    miette::set_hook(Box::new(|_| {
-        Box::new(MietteHandlerOpts::new().tab_width(2).build())
-    }))
-    .expect("miette options setup");
+    let config = Configuration::from(cli.clone());
+
+    let working_dir = update_working_dir(cli.working_dir);
 
     if let Some(source) = cli.source {
-        let content = std::fs::read_to_string(&source)?;
-        let name = source.to_string_lossy().deref().to_string();
-        let source = Source::new(&content, &name);
-        exit(handle_source(source) as i32)
+        exit(run(source, working_dir, config) as i32)
+    }
+    repl(config);
+}
+
+fn update_working_dir(working_dir: Option<PathBuf>) -> PathBuf {
+    let current_dir =
+        std::env::current_dir().expect("Unable to retrieve current working directory.");
+
+    let mut working_dir = working_dir.unwrap_or(current_dir.clone());
+    if !working_dir.is_absolute() {
+        let mut absolute_wd = current_dir;
+        absolute_wd.push(working_dir);
+        working_dir = absolute_wd;
     }
-    prompt();
-    Ok(())
+
+    if !working_dir.exists() {
+        eprintln!(
+            "working directory {} does not exists",
+            working_dir.to_string_lossy().deref()
+        );
+        exit(1);
+    }
+
+    std::env::set_current_dir(working_dir.clone()).expect("could not set current dir");
+    working_dir
+}
+
+/// https://github.com/zesterer/ariadne/issues/51
+pub fn fix_ariadne_report(report_str: String) -> String {
```

Comment:
Given that you resolved this issue, this workaround is no longer necessary, and we can wait for a proper release of `ariadne`.